json/src/spec/clique.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Clique params deserialization.
+
+use uint::Uint;
+
+/// Clique params deserialization.
+#[derive(Debug, PartialEq, Deserialize)]
+pub struct CliqueParams {
+	pub period: Option<Uint>,
+	pub epoch: Option<Uint>
```

Comment:
annotate with `#[serde(default, deserialize_with="uint::validate_optional_non_zero")]` to ensure the period is not 0

Reply:
i changed to NonZeroU64 now